src/Compilers/CSharp/Test/Symbol/Symbols/Retargeting/RetargetingTests.cs

```diff
@@ -750,6 +750,250 @@ class C1<T>
+                param2Consistent: (typeConsistent: false, refModConsistent: true, typeModConsistent: true));
+
+            TestFunctionPointerRetargetingSignature(
+                "method class [Con]C modopt([Con]C) & modopt([Con]C) *(class [Con]C modopt([Con]C) & modopt([Con]C), class [Con]C modopt([Con]C) & modopt([Ret]R))",
+                "delegate*<ref C, ref C, ref C>",
+                returnConsistent: (typeConsistent: true, refModConsistent: true, typeModConsistent: true),
+                param1Consistent: (typeConsistent: true, refModConsistent: true, typeModConsistent: true),
+                param2Consistent: (typeConsistent: true, refModConsistent: false, typeModConsistent: true));
+
+            TestFunctionPointerRetargetingSignature(
+                "method class [Con]C modopt([Con]C) & modopt([Con]C) *(class [Con]C modopt([Con]C) & modopt([Con]C), class [Con]C modopt([Ret]R) & modopt([Con]C))",
+                "delegate*<ref C, ref C, ref C>",
+                returnConsistent: (typeConsistent: true, refModConsistent: true, typeModConsistent: true),
+                param1Consistent: (typeConsistent: true, refModConsistent: true, typeModConsistent: true),
+                param2Consistent: (typeConsistent: true, refModConsistent: true, typeModConsistent: false));
+        }
+
+        private void TestFunctionPointerRetargetingSignature(
+            string ilSignature,
+            string overriddenSignature,
+            (bool typeConsistent, bool refModConsistent, bool typeModConsistent) returnConsistent,
+            (bool typeConsistent, bool refModConsistent, bool typeModConsistent) param1Consistent,
+            (bool typeConsistent, bool refModConsistent, bool typeModConsistent) param2Consistent)
+        {
+            var (retargetedAssembly1, retargetedAssembly2, consistentAssembly, originalComp, retargetedComp) = getFunctionPointerRetargetingDefinitions(ilSignature, overriddenSignature);
+
+            var mOriginal = getMethodSymbol(originalComp);
+            var mRetargeted = getMethodSymbol(retargetedComp);
+
+            Assert.IsType<RetargetingAssemblySymbol>(mRetargeted.ContainingAssembly);
+            Assert.NotSame(originalComp.Assembly, mRetargeted.ContainingAssembly);
+            Assert.NotSame(retargetedAssembly1, retargetedAssembly2);
+            Assert.Same(originalComp.Assembly, ((RetargetingAssemblySymbol)mRetargeted.ContainingAssembly).UnderlyingAssembly);
+
+            var ptrOriginal = (FunctionPointerTypeSymbol)mOriginal.ReturnType;
+            var ptrRetargeted = (FunctionPointerTypeSymbol)mRetargeted.ReturnType;
+
+            FunctionPointerUtilities.CommonVerifyFunctionPointer(ptrOriginal);
+            FunctionPointerUtilities.CommonVerifyFunctionPointer(ptrRetargeted);
+
+            assert(returnConsistent.typeConsistent,
+                   ptrOriginal.Signature.ReturnType.ContainingAssembly,
+                   ptrRetargeted.Signature.ReturnType.ContainingAssembly);
```

Comment:
>ptrRetargeted.Signature.ReturnType [](start = 19, length = 34)

Consider strengthening the test and asserting that none of the types (including modifiers) in the retargeted signature are error types.